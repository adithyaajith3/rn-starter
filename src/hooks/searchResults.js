import React, { useEffect, useState } from 'react';
import zomato from '../api/zomato';

export default () => {
    const [results, setResults] = useState([]);
    const [errorMessage, setErrorMessage] = useState('');
    useEffect(() => {
        searchApi('chinese')
    }, []);

    const searchApi = async (term) => {
        try{
            // const response = await zomato.get('/search', {
            //     query: {
            //         entity_id: 4,
            //         entity_type: 'city',
            //         q: {term},
            //         count: 10
            //     }
            // });
            const response = await zomato.get(`/search?entity_id=4&entity_type=city&q=${term}`);
            // console.log(`resto ${term}`, response.data.restaurants);
            setResults(response.data.restaurants);
        }
        catch (e) {
            console.log(e);
            setErrorMessage('Something went wrong');
        }
    }

    return [searchApi, results, errorMessage]
}