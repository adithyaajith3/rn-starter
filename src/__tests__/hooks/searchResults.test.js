import React from 'react';
import searchResults from '../../hooks/searchResults';

import renderer from 'react-test-renderer';

it('will check the matchers and pass', () => {
    const tree = renderer.create(<searchResults />).toJSON();
    expect(tree).toMatchSnapshot({
        
    })
    // const [searchApi, results, errorMessage] = searchResults();
    
  });
  