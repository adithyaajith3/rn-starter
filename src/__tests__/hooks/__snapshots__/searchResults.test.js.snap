// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will check the matchers and pass 1`] = `
Object {
  "children": null,
  "props": Object {},
  "type": "searchResults",
}
`;
